build/mac/BUILD.gn

```diff
@@ -13,55 +13,43 @@ import("//chrome/version.gni")
 import("//components/crx_file/crx3.gni")
 import("//third_party/widevine/cdm/widevine.gni")
 
-packaging_dir = "$brave_packaging_base_dir/" +
+packaging_dir = "$root_out_dir/" +
                 string_replace("$chrome_product_full_name Packaging", " ", "_")
+packaging_out_dir = "$root_out_dir/packaged"
 
 unsigned_app_path = "$root_out_dir/$brave_exe"
-if (is_universal_binary) {
-  arm64_arch_app_path = unsigned_app_path
-  x64_arch_app_path = "$root_out_dir/x64/$brave_exe"
-  unsigned_app_path = "$brave_packaging_base_dir/$brave_exe"
-}
-
-unsigned_dmg_path = "$root_out_dir/$mac_unsigned_output_prefix/$brave_dmg"
-unsigned_pkg_path = "$root_out_dir/$mac_unsigned_output_prefix/$brave_pkg"
-
-signed_app_path =
-    "$root_out_dir/$mac_signing_output_prefix/$brave_channel/$brave_exe"
-if (is_release_channel) {
-  signed_app_path = "$root_out_dir/$mac_signing_output_prefix/stable/$brave_exe"
-}
-
-signed_dmg_path = "$root_out_dir/$mac_signed_output_prefix/$brave_dmg"
-signed_pkg_path = "$root_out_dir/$mac_signed_output_prefix/$brave_pkg"
-
-app_path = signed_app_path
-if (skip_signing) {
-  app_path = unsigned_app_path
-}
-
-notarized_dmg_path = "$root_out_dir/$mac_notarized_output_prefix/$brave_dmg"
-notarized_pkg_path = "$root_out_dir/$mac_notarized_output_prefix/$brave_pkg"
 
 target_dmg_path = "$root_out_dir/$brave_dmg"
 
+upstream_packaged_basename =
+    "$packaging_out_dir/" + string_replace(chrome_product_full_name, " ", "") +
+    "-$chrome_version_full"
+upstream_dmg_path = "$upstream_packaged_basename.dmg"
+upstream_pkg_path = "$upstream_packaged_basename.pkg"
+upstream_zip_path = "$upstream_packaged_basename.zip"
+upstream_crx_path = "$upstream_packaged_basename.crx3"
+
 if (is_universal_binary) {
+  x64_arch_app_path = "$root_out_dir/x64/$brave_exe"
```

Comment:
I think we should assert `!is_universal_binary || (target_cpu == "arm64" && is_universal_binary)`, because our universalizer call is hardwired with external x64 build and in-tree arm64 build if I understood this correctly.

Reply:
Yes, good point. I added `assert(target_cpu == "arm64")` to `if (is_universal_binary) { ... }`, which I think is logically equivalent to your suggestion, in d58f93c600e382ce044f8b8712fc08eeb98ddd57.

Reply:
`assert(target_cpu == "arm64", "Universal binary builds require target_cpu to be arm64, with x64 binaries provided externally")`?